core/primitives/src/merkle.rs

```diff
@@ -199,8 +199,8 @@ mod tests {
     fn test_incorrect_path() {
         let items = vec![111, 222, 333];
         let (root, paths) = merklize(&items);
-        for i in 0..items.len() {
-            assert!(!verify_path(root, &paths[(i + 1) % 3], &items[i]))
+        for (i, item) in items.iter().enumerate() {
```

Comment:
This is slightly less readable to me.

Reply:
I think it’s less Rustic to use indexing, but this isn’t relevant to the actual change in the PR so gonna revert this hunk.